### PR TITLE
build: Add script to build & minify loader

### PR DESCRIPTION
**Pull request by mydea**
This adds a simple build script for the JS Loader. This should help with two things:

* The base template is plain JS, making syntax highlighting and error finding etc. much easier
* Have a reproducible minification step

This just has to be run manually whenever the loader script is changed, the output is fully checked in etc. so nothing changes there.

I've tried the minified output in a local test app, would be great if someone else could also give it a try!

### CODE REVIEW

**Review by lforst — APPROVED** [review]

File: scripts/build-js-loader.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const {minify} = require('terser');
```
Comment (evanpurkhiser) [inline scripts/build-js-loader.js:1]: Are you able to write this script in typescript?

File: scripts/build-js-loader.js
Comment (evanpurkhiser) [inline scripts/build-js-loader.js:1]: Are you able to write this script in typescript?

File: scripts/build-js-loader.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  console.log(`Updated loader min. script template at ${loaderMinTmplPath}`);
+}
+
+function replacePlaceholders(str) {
+  return str
+    .replace('__LOADER__PUBLIC_KEY__', "'{{ publicKey|safe }}'")
+    .replace('__LOADER_SDK_URL__', "'{{ jsSdkUrl|safe }}'")
+    .replace('__LOADER__CONFIG__', '{{ config|to_json|safe }}')
+    .replace('__LOADER__IS_LAZY__', '{{ isLazy|safe|lower }}');
+}
```
Comment (evanpurkhiser) [inline scripts/build-js-loader.js:41]: Might be nice to have some comments on what is going on here

**Comment by priscilawebdev** [comment]
will this generate the script we are displaying in the onboarding? it is not clear to me what is the goal of this PR

**Review by AbhiPrasad — APPROVED** [review]
Minified loader impl looks good!

File: scripts/build-js-loader.js (hunk 2 above)
Comment (mydea) [reply scripts/build-js-loader.js:45]: I added a comment there!

File: scripts/build-js-loader.js
Comment (mydea) [reply scripts/build-js-loader.js:1]: So I set it up to have prettier & eslint. We _could_ use typescript but that would complicate the build step - we'd need to use either rollup or babel or something like this then. Right now we just use the script as-is mostly (which also means the script has to be ES5 compatible!)

**Comment by mydea** [comment]
> will this generate the script we are displaying in the onboarding? it is not clear to me what is the goal of this PR

This script generates the actual Django templates from a base .js file. The goal of this is mainly to streamline the minification of this script, as previously we did that manually e.g. online by closure compiler. This is a) manual, b) error prone, c) not reproducible very well (and actually lead to an incident last Friday).

I added a comment to the top of the script explaining the purpose of the script in detail as well!

Note: I've also checked the source script for ES5 compatibility to make sure it is valid ES5 JS.

File: yarn.lock [hunk 3]
```diff
@@ -10481,6 +10481,16 @@ terser-webpack-plugin@^5.1.3:
...
+  version "5.16.8"
+  resolved "https://registry.yarnpkg.com/terser/-/terser-5.16.8.tgz#ccde583dabe71df3f4ed02b65eb6532e0fae15d5"
+  integrity sha512-QI5g1E/ef7d+PsDifb+a6nnVgC4F22Bg6T0xrBrz6iloVB4PUkkunp6V8nzoOOZJIzjWVdAGqCdlKlhLq/TbIA==
+  dependencies:
+    "@jridgewell/source-map" "^0.3.2"
+    acorn "^8.5.0"
+    commander "^2.20.0"
+    source-map-support "~0.5.20"
+
 terser@^5.10.0, terser@^5.7.0:
```
Comment (scttcper) [inline yarn.lock:10494]: confusing which one webpack will use when building

can you run
```sh
npx yarn-deduplicate
yarn install
```

it should bring us back to only one version installed

File: scripts/build-js-loader.js
Comment (mydea) [reply scripts/build-js-loader.js:1]: OK, so i refactored it to use typescript and use `typescript` directly to compile it!

**Review by priscilawebdev — APPROVED** [review]
the types in the file `src/sentry/templates/sentry/js-sdk-loader.ts` still can be improved but it looks good to me!

**Review by AbhiPrasad — APPROVED** [review]

File: scripts/build-js-loader.js
Comment (mydea) [reply scripts/build-js-loader.js:1]: I also rewrote the build-script itself to typescript.

**Comment by mydea** [comment]
OK, so I tested this multiple times in a local test app, both minified and unminifed seem to be working as expected!

**Pull request merged**